src/main/java/com/uid2/shared/store/CloudPath.java

```diff
@@ -38,7 +58,17 @@ public String toString() {
 
         // Java's Path always converts everything to path standards of the OS it's running on.
         // The paths are actually S3 paths so should always be Linux style, so we need to convert them when on Windows
-        return isWindows ? path.replace("\\", "/") : path ;
+        if (isWindows) {
+            path = path.replace("\\", "/");
+        };
+
+        if (uriBase != null) {
+            return path.isEmpty() ?
+                    uriBase.getScheme() + "://" + uriBase.getAuthority() :
+                    uriBase.resolve("/").resolve(path).normalize().toString();
+        }
+
+        return path;
     }
 
```

Comment:
should we use/compare uriBase in equals method and hashCode method below?

Reply:
nice catch, updated.